enhancements/olm-curated-catalog.md

```diff
@@ -0,0 +1,353 @@
+---
+title: olm-curated-catalog
+
+authors:
+  - "@cdjohnson"
+
+reviewers:
+
+approvers:
+
+creation-date: 2020-03-19
+
+last-updated: 2020-03-19
+
+status: 
+
+see-also:
+  - "/enhancements/opm-mirroring.md"  
+
+replaces:
+
+superseded-by:
+---
+
+# OLM Curated Catalog
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+- Where is the best place to physically mirror images.  `opm` or `oc`?
+  - It appears that [opm-mirroring.md](opm-mirroring.md) was implmented as an OpenShift enhancment to: `oc adm catalog mirror`
+
+
+## Summary
+
+This enhancement outlines the changes necessary for `opm` to support curated catalog indexes to support distribution of portions of existing OLM catalogs for disconnected environments. This relies on the [Disconnected OLM enhancement](opm-mirroring.md).
+
+## Motivation
+
+There are two primary motivations:
+
+### Disconnected Catalog Mirroring
+Disconnected installation of OLM catalogs can be done with the following OpenShift Container Platform `oc` command:
+
+```
+$ oc adm catalog mirror quay.io/openshift/community-operators-catalog:latest disconnected-registry:5000/community-operators/
+
+```
+
+This will mirror the entire catalog to the disconnected, local registry.  As catalogs begin to include hundreds of operators, versions of operators, versions of operands and multiple architectures, it will not be feasible to mirror the entire online catalog.
+
+### Curated Catalogs for Users
+Companies would like to be able to limit the list of operators to those that are endorsed or supported by the customer and their lines of business.
```

Comment:
This is an interesting statement. I wonder if the `--filter-by-os` or `--filter-by-version` is just a small subset of the actual want here. Wouldn't it be nicer to be able to actually describe the graph declaratively for each package that you want (via a json object input or something similar)? I wonder if these configuration options will end up getting bigger and more complex over time as more features are requested.

Reply:
I read this statement as higher-level, along the lines of: "As a user, I need updates for the database operator that I selected. I don't need to mirror the other 12 database operators that are available in the same index." Some kind of white-list would work.